src/lib/storage/segment_encoding_utils.cpp

```diff
@@ -52,4 +51,22 @@ std::shared_ptr<BaseEncodedSegment> encode_segment(EncodingType encoding_type, D
   return encoder->encode(segment, data_type);
 }
 
+/**
+ * @brief Returns the vector compression type for a given compressed vector type.
+ *
+ * For the difference of the two, please take a look at compressed_vector_type.hpp.
+ */
+VectorCompressionType parent_vector_compression_type(const CompressedVectorType compressed_vector_type) {
+  switch (compressed_vector_type) {
+    case CompressedVectorType::FixedSize4ByteAligned:
+    case CompressedVectorType::FixedSize2ByteAligned:
+    case CompressedVectorType::FixedSize1ByteAligned:
+      return VectorCompressionType::FixedSizeByteAligned;
```

Comment:
Uargh. Introducing "parent types" makes #1286 even more necessary. `VectorCompressionAlgorithm` and `VectorCompressionType`?

Reply:
Copied your recommendation to the issue.